uiplib/gui/LinuxMinimizer.py

```diff
@@ -0,0 +1,113 @@
+
+    def __init__(self, settings, wallpaper, index, images):
+        """Initialize the minimizer."""
+        self.wallpaper = wallpaper
+        self.settings = settings
+        self.index = index
+        self.images = images
+        self.indicator = appindicator.Indicator.new(
+            APPINDICATOR_ID, gtk.STOCK_DIALOG_INFO,
+            appindicator.IndicatorCategory.SYSTEM_SERVICES)
+
+    # Menu items
+
+    def quit(self, source):
+        """Quit the Minimizer."""
+        notify.uninit()
+        gtk.main_quit()
+
+    def _prev(self, source):
+        """Set Previous wallpaper."""
+        notify_prev.show()
+        self.update()
+        self.index = (self.index - 1 if self.index != 0
+                      else len(self.images) - 1)
+        self.image = self.images[self.index].image_path
+        self.set_wallpaper()
+        print('Prev image')
+
+    def _next(self, source):
+        """Set Next wallpaper."""
+        notify_next.show()
+        self.update()
+        self.index = (self.index + 1) % len(self.images)
+        self.image = self.images[self.index].image_path
+        self.set_wallpaper()
+        print('Next image')
+
+    def set_wallpaper(self):
+        """Set the wallpaper which is being previewed."""
+        self.wallpaper.set(self.image)
+
+    def maximize(self, source):
+        """Restore the maximized window of UIP."""
+        print('Maximizing UIP')
+        app = MainWindow(self.settings, self.wallpaper)
+        app.run()
+
+    # Building Menu
+
+    def build_menu(self):
+        """Build the menu for status icon."""
+        self.menu = gtk.Menu()
+        self.item_next = gtk.MenuItem('Next')
+        self.item_next.connect('activate', self._next)
+        self.menu.append(self.item_next)
+
+        self.item_prev = gtk.MenuItem('Previous')
+        self.item_prev.connect('activate', self._prev)
+        self.menu.append(self.item_prev)
+
+        self.item_max = gtk.MenuItem('Maximize')
+        self.item_max.connect('activate', self.maximize)
+        self.menu.append(self.item_max)
+
+        self.item_quit = gtk.MenuItem('Quit')
+        self.item_quit.connect('activate', quit)
+        self.menu.append(self.item_quit)
+
+        self.menu.show_all()
+        return self.menu
+
+    def run(self):
+        """Method to run the status icon."""
+        self.indicator.set_status(appindicator.IndicatorStatus.ACTIVE)
+        self.indicator.set_menu(self.build_menu())
+        notify.init(APPINDICATOR_ID)
+        gtk.main()
+
+    def update(self):
+        """Method to get images from directory."""
+        print("Updating Image List")
+        src_directory = self.settings['fav-pics-folder']
+        src_files = os.listdir(src_directory)
+        for file_name in src_files:
+            full_file_name = os.path.join(src_directory, file_name)
+            if (os.path.isfile(full_file_name)):
+                copy(full_file_name, self.settings['pics-folder'])
```

Comment:
@Aniq55 at what stage? we would always like to make changes in the master that's better for the software right? it's ohk if it breaks other things, we'll fix those too? plus i'm not asking you to make a new file anyway, just move these to, `utils.py`